### PR TITLE
CT-2895 Third party requester name is optional field

### PR DESCRIPTION
**Pull request by njpreddy**
## Description
CT-2165 Third party requester name is optional field

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [x] (6) No TODO's without new ticket numbers
* [x] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Screenshots
**1. Errors when all third party required fields empty:**

all-required-empty

**2. Errors when only third party relationship name field completed:**

relationship-only-entered

**3. Errors when only third party representative name field completed:**

representative-only-entered

**4. Errors when only third party company name field completed:**

company-only-entered

### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-2895

### CODE REVIEW

File: app/models/case/sar/offender.rb [hunk 1]
```diff
@@ -46,7 +46,8 @@ def searchable_fields_and_ranks
                  third_party: :boolean,
                  third_party_reference: :string,
                  third_party_company_name: :string,
-                 late_team_id: :integer
+                 late_team_id: :integer,
+                 third_party_name: :string
```
Comment (Polsonby) [inline app/models/case/sar/offender.rb:50]: Good spot 👍

File: app/models/case/sar/offender.rb [hunk 2]
```diff
@@ -72,11 +73,11 @@ def searchable_fields_and_ranks
   has_many :data_requests, dependent: :destroy, foreign_key: :case_id
   accepts_nested_attributes_for :data_requests
 
-  validates :third_party, inclusion: { in: [true, false], message: "can't be blank" }
+  validates :third_party,          inclusion: { in: [true, false], message: "can't be blank" }
   validates :flag_as_high_profile, inclusion: { in: [true, false], message: "can't be blank" }
 
-  validates :name, presence: true, if: -> { third_party }
   validates :third_party_relationship, presence: true, if: -> { third_party }
+  # validates :third_party_company_name, presence: true, if: -> { third_party }
```
Comment (Polsonby) [inline app/models/case/sar/offender.rb:80]: shall we remove the commented out line here?

**Review by Polsonby — APPROVED** [review]
LGTM :) 

Great job :) 

Maybe remove the commented out line before merging?

File: app/models/case/sar/offender.rb (hunk 2 above)
Comment (njpreddy) [reply app/models/case/sar/offender.rb:80]: Oh yes! Doh!

**Pull request merged**